mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala

```diff
@@ -142,5 +142,13 @@ object AssociationRules {
     def javaConsequent: java.util.List[Item] = {
       consequent.toList.asJava
     }
+
+    /**
+     * @return [antecedent] => [consequent]: [confidence]
+     */
+    @Since("1.6.0")
+    override def toString = {
+      s"${javaAntecedent} => ${javaConsequent}: ${confidence}"
```

Comment:
Using `javaAntecedent` is hacky. We should use `antecedent.mkString("{", ",", "}")` instead. Same for `consequent`.